### ISSUE TITLE
Wrong checkout version Moya, AlamofireImage, ReactiveCocoa

### ISSUE THREAD

**Issue by tudemon** [Carthage/Carthage#2322]
* `carthage version`: 0.27.0
* `xcodebuild -version`: 9.2
* Are you using `--no-build`? NO
* Are you using `--no-use-binaries`? YES
* Are you using `--use-submodules`? NO
* Are you using `--cache-builds`? YES 

I want to install Moya version 10.0.1, AlamofireImage 3.3.0 and ReactiveCocoa version 7.1.0. But when I add these 3 dependencies at the same time, the versions is NOT as expected.

**Cartfile**
```
# Moya
github "Moya/Moya"

# AlamofireImage
github "Alamofire/AlamofireImage"
```

**Carthage Output**
```
*** Fetching Moya
*** Fetching AlamofireImage
*** Fetching Alamofire
*** Fetching Result
*** Fetching ReactiveSwift
*** Fetching RxSwift
*** Checking out RxSwift at "4.1.1"
*** Checking out Result at "3.2.4"
*** Checking out Alamofire at "4.6.0"
*** Checking out ReactiveSwift at "2.0.1"
*** Checking out AlamofireImage at "3.3.0"
*** Checking out Moya at "10.0.1"
```

---> Moya : 10.0.1, AlamofireImage : 3.3.0 👍 

**Cartfile**
```
# Moya
github "Moya/Moya"

# ReactiveCocoa
github "ReactiveCocoa/ReactiveCocoa" ~> 5.0.4

# AlamofireImage
github "Alamofire/AlamofireImage"
```

**Carthage Output**
```
*** Fetching Moya
*** Fetching AlamofireImage
*** Fetching ReactiveCocoa
*** Fetching ReactiveSwift
*** Fetching Alamofire
*** Fetching Result
*** Fetching RxSwift
*** Checking out ReactiveCocoa at "5.0.4"
*** Checking out RxSwift at "3.6.1"
*** Checking out Moya at "8.0.5"
*** Checking out Alamofire at "4.6.0"
*** Checking out ReactiveSwift at "1.1.5"
*** Checking out AlamofireImage at "3.3.0"
*** Checking out Result at "3.2.4"
```

---> Moya : **8.0.5**, AlamofireImage : 3.3.0, ReactiveCocoa : 5.0.4 👎  

**Cartfile**
```
# Moya
github "Moya/Moya"

# ReactiveCocoa
github "ReactiveCocoa/ReactiveCocoa"

github "Alamofire/AlamofireImage"
```

**Carthage Output**
```
*** Fetching Moya
*** Fetching AlamofireImage
*** Fetching ReactiveCocoa
*** Fetching Result
*** Fetching Alamofire
*** Fetching ReactiveSwift
*** Fetching RxSwift
*** Checking out ReactiveCocoa at "7.1.0"
*** Checking out Result at "3.2.4"
*** Checking out Alamofire at "4.6.0"
*** Checking out ReactiveSwift at "3.1.0"
*** Checking out AlamofireImage at "3.3.0"
*** Checking out Moya at "3.0.0"
```

---> Moya : **3.0.0**, AlamofireImage : 3.3.0, ReactiveCocoa : 7.1.0 👎

**Comment by tudemon** [comment]
@mdiep please help T_T

**Comment by tudemon** [comment]
I resolved with deleting cache folder 
`rm -rf ~/Library/Caches/org.carthage.CarthageKit`

**Comment by blender** [comment]
@tudemon in your second example you are locking

```
"ReactiveCocoa/ReactiveCocoa" ~> 5.0.4
```

I checked, as you should have done, and [ReactiveCocoa 5.0.4 is compatible with ReactiveSwift 1.1 ](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/5.0.4/Cartfile). This in turns pulls a version of Moya which is compatible with ReactiveSwift 1.1. This version is Moya : 8.0.5

This works as expected.

In your last example I could not find a good reason why this is happening and even when requiring the following Carthage still does the wrong thing breaking requirements. This is definitely a bug.

```
github "Moya/Moya" ~> 10.0.0

# ReactiveCocoa
github "ReactiveCocoa/ReactiveCocoa" ~> 7.1.0

github "Alamofire/AlamofireImage" ~> 4.6.0
```

**Comment by ikesyo** [comment]
FYI trying the `--new-resolver` option which was added in 0.27.0 may be a help.

**Comment by leontiy** [comment]
@mdiep has this issue been resolved?

I keep having similar issues with Carthage's dependency resolution since last year. At one time, updating Carthage helped. Another time I realised that dependencies were incompatible and Carthage would resolve with a seemingly random version of RAC without any error messages. Today, I've encountered this problem again. Here's my Cartfile:

```
github "ReactiveCocoa/ReactiveSwift" ~> 6.0
github "ReactiveCocoa/ReactiveCocoa" ~> 9
```

Here's what Carthage checks out with `carthage update --use-submodules --no-build`. I've also tried `--new-resolver` to no avail.

```
*** Checking out Quick at "v2.0.0"
*** Checking out ReactiveCocoa at "v3.0-beta.9"
*** Checking out Box at "1.2.2"
*** Checking out Nimble at "v8.0.1"
*** Checking out Result at "0.4.5"
*** Checking out ReactiveSwift at "6.0.0"
```
